rust/kernel/file.rs

```diff
@@ -6,36 +6,37 @@
 //! [`include/linux/file.h`](../../../../include/linux/file.h)
 
 use crate::bindings;
+use core::ptr::NonNull;
 
 /// Wraps the kernel's `struct file`.
 ///
 /// # Invariants
 ///
-/// The pointer [`File::ptr`] is non-null and valid.
+/// The pointer [`File::ptr`] is valid.
 pub struct File {
-    pub(crate) ptr: *const bindings::file,
+    pub(crate) ptr: NonNull<bindings::file>,
 }
 
 impl File {
     /// Constructs a new [`struct file`] wrapper.
     ///
     /// # Safety
     ///
-    /// The pointer `ptr` must be non-null and valid for the lifetime of the object.
-    pub(crate) unsafe fn from_ptr(ptr: *const bindings::file) -> File {
+    /// The pointer `ptr` must be valid for the lifetime of the object.
+    pub(crate) unsafe fn from_ptr(ptr: NonNull<bindings::file>) -> File {
```

Comment:
Usually `from_ptr` will accept a raw pointer instead of `NonNull` -- it's unsafe and up to the caller to guarantee the validity anyway.